dali/python/backend_impl.cc

```diff
@@ -1623,10 +1623,19 @@ PYBIND11_MODULE(backend_impl, m) {
           if (is_cpu_data) {
             FeedPipeline<CPUBackend>(p, name, list, AccessOrder::host(), true);
           } else {
-            cudaStream_t stream = cuda_stream.is_none()
-                                ? UserStream::Get()->GetStream(list[0].cast<Tensor<GPUBackend>&>())
-                                : static_cast<cudaStream_t>(ctypes_void_ptr(cuda_stream));
-            FeedPipeline<GPUBackend>(p, name, list, stream, cuda_stream.is_none(), use_copy_kernel);
+            int device_id = p->device_id();
+            cudaStream_t stream = 0;
+            if (!cuda_stream.is_none())
+              stream = static_cast<cudaStream_t>(ctypes_void_ptr(cuda_stream));
+
+            if (!list.empty()) {
```

Comment:
If the list is empty L1622 would fail anyway. Maybe we should enforce it to be not empty?

Reply:
Oh, I haven't looked that high up. I guess it should be adjusted there (and tested?). And... no, I think it's legitimate to have an empty array - perhaps not now, but with totally variable batch size (split/merge) it's quite reasonable to expect empty batches at some points in the pipeline.

Reply:
Fixed - but it cannot be tested right now, empty batch causes problems elsewhere and it's definitely beyond the scope of this task to fix it.

Reply:
Now this check is not needed.